swarm/network/stream/stream.go

```diff
@@ -666,17 +666,16 @@ func peerStreamIntervalsKey(p *Peer, s Stream) string {
 	return p.ID().String() + s.String()
 }
 
-func (c client) AddInterval(start, end uint64) (err error) {
-	i := &intervals.Intervals{}
-	err = c.intervalsStore.Get(c.intervalsKey, i)
-	if err != nil {
+func (c *client) AddInterval(start, end uint64) (err error) {
+	i := new(intervals.Intervals)
+	if err = c.intervalsStore.Get(c.intervalsKey, i); err != nil {
 		return err
 	}
 	i.Add(start, end)
 	return c.intervalsStore.Put(c.intervalsKey, i)
 }
 
-func (c client) NextInterval() (start, end uint64, err error) {
+func (c *client) NextInterval() (start, end uint64, err error) {
 	i := &intervals.Intervals{}
```

Comment:
As intervals.Intervals allocation is changed in AddIntervals to use new, and the Get function call is inside if, should this be changed here, also?

Reply:
Nice! :1st_place_medal:  
Yup. Let's keep it consistent.  :)

Reply:
fixed

Reply:
Cool, even though, I do not see why this change is needed to solve the issue, but if you like to make it better, great. There are other examples, like https://github.com/ethereum/go-ethereum/blob/master/swarm/network/stream/stream.go#L112, https://github.com/ethereum/go-ethereum/blob/master/swarm/network/stream/peer.go#L329.

Reply:
Bot allocations a re fine for me, it is good to be consistent as much as possible. Shouldn't the err assignment a line bellow be under the if statement just as it is changed in AddInterval?

Reply:
I strongly prefer consistency, too. My eyes skipped that other `err` statement. Too late I guess. :(